pkg/controller/standard.oam.dev/v1alpha1/rollout/handler.go

```diff
@@ -219,3 +237,44 @@ func (h *handler) isRolloutModified(rollout v1alpha1.Rollout) bool {
 			(rollout.Spec.RolloutPlan.TargetSize != nil && rollout.Status.RolloutTargetSize != -1 &&
 				rollout.Status.RolloutTargetSize != *rollout.Spec.RolloutPlan.TargetSize))
 }
+
+func (h *handler) recordeWorkloadInResourceTracker(ctx context.Context) error {
+	var resourceTrackerName string
+	for _, reference := range h.rollout.OwnerReferences {
+		if reference.Kind == v1beta1.ResourceTrackerKind && reference.APIVersion == v1beta1.SchemeGroupVersion.String() {
+			resourceTrackerName = reference.Name
+		}
+	}
+	if len(resourceTrackerName) == 0 {
+		// rollout isn't created by application
+		return nil
+	}
+	rt := v1beta1.ResourceTracker{}
+	if err := h.Get(ctx, types.NamespacedName{Name: resourceTrackerName}, &rt); err != nil {
+		klog.Errorf("rollout recordeWorkloadInResourceTracker error to get resourceTracker namespace:%s, name: %s", h.rollout.Namespace, h.rollout.Name)
+		return err
+	}
+	recordedWorkload := corev1.ObjectReference{
+		APIVersion: h.targetWorkload.GetAPIVersion(),
+		Kind:       h.targetWorkload.GetKind(),
+		UID:        h.targetWorkload.GetUID(),
+		Namespace:  h.targetWorkload.GetNamespace(),
+		Name:       h.targetWorkload.GetName(),
+	}
+	rt.Status.TrackedResources = append(rt.Status.TrackedResources, recordedWorkload)
+	if err := h.Status().Update(ctx, &rt); err != nil {
+		klog.Errorf("rollout recordeWorkloadInResourceTracker error recorde workload rollout namespace:%s, name: %s", h.rollout.Namespace, h.rollout.Name)
```

Comment:
```suggestion
		klog.Errorf("fail to update resource tracker for rollout record workload namespace:%s, name: %s", h.rollout.Namespace, h.rollout.Name)
```

这些error信息你要考虑debug的人能不能看懂